src/HealthChecks.MySql/MySqlHealthCheck.cs

```diff
@@ -12,8 +12,11 @@ public class MySqlHealthCheck : IHealthCheck
 
     public MySqlHealthCheck(MySqlHealthCheckOptions options)
     {
-        Guard.ThrowIfNull(options.ConnectionString, true);
-        Guard.ThrowIfNull(options.CommandText, true);
+        Guard.ThrowIfNull(options);
+        if (options.DataSource is null && options.ConnectionString is null)
+            throw new InvalidOperationException("One of options.DataSource or options.ConnectionString must be specified.");
+        if (options.DataSource is not null && options.ConnectionString is not null)
+            throw new InvalidOperationException("Only one of options.DataSource or options.ConnectionString must be specified.");
         _options = options;

```

Comment:
We don't need these checks anymore, as the public ctors ensure that the state is always valid.

```suggestion
        _options = Guard.ThrowIfNull(options);
```

Reply:
For some reason I can't apply this suggestion. I am going to merge it now since it's not blocking

Reply:
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2129